### ISSUE TITLE
prrte (master) not forwarding local_size - and maybe other attributes - to processes when using ALPS RAS/PLM

### ISSUE THREAD

**Issue by hppritcha** [openpmix/prrte#646]
After some digging, it appears that somethings not right with prrte when used in an alps environment.  One of the simple examples illustrates the problem.

When using prrte on an Cray XC30 using SLURM the test runs correctly:
```
Client 252650: Running
Client ns prte-jupiter26462.2 rank 0 pid 252650: Running
Client prte-jupiter26462.2:0 universe size 8
Client prte-jupiter26462.2:0 num procs 1
Client prte-jupiter26462.2:0 num local procs 1
Client ns prte-jupiter26462.2 rank 0: PMIx_Get prte-jupiter26462.2-0-local returned correct
Client ns prte-jupiter26462.2 rank 0: PMIx_Get prte-jupiter26462.2-0-remote returned correct
Client ns prte-jupiter26462.2 rank 0: Finalizing
Client ns prte-jupiter26462.2 rank 0:PMIx_Finalize successfully completed
```
However, when run on a Cray XC30 using ALPS+PBSpro the get of ```local_size``` fails:
```
Client 24276: Running
Client ns prte-crystal7254.2 rank 0 pid 24276: Running
Client prte-crystal7254.2:0 universe size 32
Client prte-crystal7254.2:0 num procs 1
Client ns prte-crystal7254.2 rank 0: PMIx_Get local size failed: -46
Client ns prte-crystal7254.2 rank 0: Finalizing
Client ns prte-crystal7254.2 rank 0:PMIx_Finalize successfully completed
```
This doesn't appear to be a pmix problem, at least not directly.  In the test setup, I built pmix once and am using it on both systems.  pmix itself doesn't have any alps specific code either.  It appears likely that something is either going on in the 
```prte_pmix_server_register_nspace``` method or else in the retrivial of this information on the client side.

Note on both systems the ALPS odls is being used.

**Comment by hppritcha** [comment]
If one gets gds to dump out debug stuff in the client shell it appears that somehow some keys are mysteriously missing when run under alps vs slurm.
When using slurm:
```
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.bfrops.mod look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.node.size look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.lpeers look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.lldr look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.local.size look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key nid00060 look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.appnum look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.app.size look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.aldr look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.wdir look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.app.argv look for key pmix.dbg.init
[nid00060:05826] dstore_base.c:2237:_dstore_fetch: for rank prte-jupiter22241.2:4294967294, skip key pmix.pdata look for key pmix.dbg.init
```

but when running with ALPS/Pbspro or cobalt:
```
nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.bfrops.mod look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key 84 look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.appnum look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.app.size look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.aldr look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.wdir look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.app.argv look for key pmix.dbg.init
[nid00084:14708] dstore_base.c:2237:_dstore_fetch: for rank prte-crystal227773.2:4294967294, skip key pmix.pdata look for key pmix.dbg.init
```

**Comment by hppritcha** [comment]
okay found issue.

**Comment by hppritcha** [comment]
this was fixed via #648
